crates/evm-prover/src/storage/storage.rs

```diff
@@ -0,0 +1,300 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use celestia_types::nmt::Namespace;
+use evm_exec_types::{BlockExecOutput, BlockRangeExecOutput};
+use rocksdb::{ColumnFamily, ColumnFamilyDescriptor, Options, DB};
+use serde::{Deserialize, Serialize};
+use sp1_sdk::SP1ProofWithPublicValues;
+use std::path::Path;
+use std::sync::Arc;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ProofStorageError {
+    #[error("Database error: {0}")]
+    Database(#[from] rocksdb::Error),
+    #[error("Serialization error: {0}")]
+    Serialization(#[from] bincode::Error),
+    #[error("General error: {0}")]
+    General(#[from] anyhow::Error),
+    #[error("Proof not found for height: {0}")]
+    ProofNotFound(u64),
+    #[error("Range proof not found for range: {0}-{1}")]
+    RangeProofNotFound(u64, u64),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct StoredBlockProof {
+    pub celestia_height: u64,
+    pub celestia_header_hash: [u8; 32],
+    pub evm_height: u64,
+    pub evm_state_root: [u8; 32],
+    pub namespace: Namespace,
+    pub proof_data: Vec<u8>,
+    pub public_values: Vec<u8>,
+    pub created_at: u64,
+}
```

Comment:
I think we can reduce this as many of the fields are duplicates since they are encoded within `public_values`:

```suggestion
pub struct StoredBlockProof {
    pub celestia_height: u64,
    pub proof_data: Vec<u8>,
    pub public_values: Vec<u8>,
    pub created_at: u64,
}
```